src/HashedExpression/Value.hs

```diff
@@ -5,27 +5,38 @@ import Data.Map (Map, fromList)
 import qualified Data.Map as Map
 import HashedExpression.Internal.Expression
 
+-- | The name of a dataset within an HDF5 file
 type Dataset = String
 
+-- | The types of data files supported by HashedExpression
 data DataFile
-  = TXT FilePath
-  | -- C support hdf5, we want npy too but haven't found any npy reader library for C
-    -- file path to the data file from your solver
-    HDF5 FilePath Dataset -- HDF5 requires the name of the data set in the data file
+  = -- | File path to plaintext data file from your solver
+    TXT FilePath
+  | -- | HDF5 requires the name of the data set in the data file
+    --   C supports hdf5, we want npy too but haven't found any npy reader library for C
+    HDF5 FilePath Dataset
   deriving (Eq, Show, Ord)
 
+-- | A value in a Symphony problem.
 data Val
-  = VScalar Double
-  | V1D (Array Int Double)
-  | V2D (Array (Int, Int) Double)
-  | V3D (Array (Int, Int, Int) Double)
-  | VFile DataFile
-  | VNum Double
+  = -- | Constant scalar double value
+    VScalar Double
+  | -- | One-dimensional double array
+    V1D (Array Int Double)
+  | -- | Two-dimensional double array
+    V2D (Array (Int, Int) Double)
+  | -- | Three-dimensional double array
+    V3D (Array (Int, Int, Int) Double)
+  | -- | Path to a data file
+    VFile DataFile
+  | -- | TODO: Nhan, what is this?
```

Comment:
This is a literal double value that fit all kind of dimensions.

Reply:
Oh okay, so it's used to create a structure with every element equal to the same value?

Reply:
Like a 100-by-100 matrix of 2s for example?

Reply:
Yes!, so if you remove this TODO we can merge it.

Reply:
Will do!